client/core/core.go

```diff
@@ -1804,23 +1892,30 @@ func (c *Core) setWalletPassword(wallet *xcWallet, newPW []byte, crypter encrypt
 	// TODO: find a way to verify that the wallet actually is unencrypted or
 	// otherwise does not require a password. Perhaps an
 	// asset.Wallet.RequiresPassword wallet method?
-	if newPasswordSet {
-		// Encrypt password if it's not an empty string
-		encNewPW, err := crypter.Encrypt(newPW)
-		if err != nil {
-			return newError(encryptionErr, "encryption error: %v", err)
-		}
-		err = wallet.Wallet.Unlock(string(newPW))
-		if err != nil {
-			return newError(authErr,
-				"setWalletPassword unlocking wallet error, is the new password correct?: %v", err)
+	updateWalletPW := func(w *xcWallet) error {
+		if newPasswordSet {
+			// Encrypt password if it's not an empty string.
```

Comment:
Hrmm you are right regarding no utility, i do find a bit more readable that way, instead of have the logic inside the for
loop.

Another option would be to define it as unexported `Core` func, what do you think ?

Reply:
> inside the for loop

Did you mean the `if` block, or were you thinking of the `for` loop in `updateWalletsEncPW`? If you meant the `if` block, it's still there, just inside of a function now. I don't really see how that makes it more readable.

Reply:
I was thinking of the `for` in `updateWalletsEncPW` but it's not that bad having the logic there, moving it

Reply:
Updating this one, both inline funcs are `updateWalletPW` this why I got confused 🙈.

Reply:
Done